Sources_v3/StreamChatUI/ChatChannelList/ChatChannelView.swift

```diff
@@ -0,0 +1,140 @@
+        get { channelNameLabel.textColor }
+        set { channelNameLabel.textColor = newValue }
+    }
+    
+    @objc public dynamic var channelAvatarViewBackgroundColor: UIColor? {
+        get { channelAvatarView.backgroundColor }
+        set { channelAvatarView.backgroundColor = newValue }
+    }
+    
+    @objc public dynamic var separatorColor: UIColor? {
+        get { separatorView.backgroundColor }
+        set { separatorView.backgroundColor = newValue }
+    }
+    
+    // MARK: - Init
+    
+    public required init(
+        channel: _ChatChannel<ExtraData>? = nil,
+        uiConfig: UIConfig<ExtraData> = .default
+    ) {
+        self.channel = channel
+        self.uiConfig = uiConfig
+        
+        super.init(frame: .zero)
+        
+        commonInit()
+    }
+    
+    public required init?(coder: NSCoder) {
+        self.uiConfig = .default
+        self.channel = nil
+        
+        super.init(coder: coder)
+        
+        commonInit()
+    }
+    
+    public func commonInit() {
+        setupLayout()
+        setupAppearance()
+        updateAppearance()
+    }
+
+    // MARK: - Public
+    
+    open func setupAppearance() {
+        channelNameLabel.textColor = .black
+        channelNameLabel.numberOfLines = 1
+        channelNameLabel.font = UIFontMetrics(forTextStyle: .subheadline)
+            .scaledFont(for: .boldSystemFont(ofSize: UIFont.systemFontSize))
+        
+        separatorView.backgroundColor = .lightGray
+    }
+    
+    open func setupLayout() {
+        addSubview(container)
+
+        container.centerStackView.isHidden = false
+        container.centerStackView.spacing = 8
+        container.centerStackView.alignment = .center
+        container.centerStackView.distribution = .fill
+        container.centerStackView.addArrangedSubview(channelAvatarView)
+        container.centerStackView.addArrangedSubview(channelNameLabel)
+        
+        addSubview(separatorView)
+
+        NSLayoutConstraint.activate([
+            // Layout container.
+            container.leadingAnchor.constraint(equalTo: leadingAnchor, constant: 8),
+            container.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -8),
+            container.topAnchor.constraint(equalTo: topAnchor, constant: 12),
+            container.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -12),
+            
+            // Layout avatar.
+            channelAvatarView.widthAnchor.constraint(equalToConstant: 40),
+            
+            // Layout separator.
+            separatorView.heightAnchor.constraint(equalToConstant: 1 / UIScreen.main.scale),
+            separatorView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            separatorView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            separatorView.bottomAnchor.constraint(equalTo: bottomAnchor)
+        ])
+    }
+    
+    open func updateAppearance() {
```

Comment:
`updateContent()` ? -> figure this out

Reply:
renamed in 6c7a353a11bc8b3283af7b7911fe3f83c7a4b405